src/leader.py

```diff
@@ -142,14 +143,45 @@ def lead(self):
         Moves the leader vehicle based on controller input.
         """
         code, position = self._controller.get_input()
-        if code == 'right_trigger':
-            self.set_speed(position)
-            self.drive()
-        if code == 'left_trigger':
-            self.set_speed(position)
-            self.reverse()
-        if code == 'dpad_left_right' or code == 'left_stick_x':
-            self.turn(code, position)
+        if code == 'left_trigger' or code == 'right_trigger':
+            self._mode = 'normal'
+
+        elif code == 'x' and position == 1:
+            if self._mode == 'drives_in_circle':
+                self._mode = 'normal'
+            else:
+                self._mode = 'drives_in_circle'
+
+        elif code == 'y' and position == 1:
+            if self._mode == 'drive_wide_winding':
+                self._mode = 'normal'
+            else:
+                self._mode = 'drive_wide_winding'
+
+        elif code == 'b' and position == 1:
+            if self._mode == 'drive_winding':
+                self._mode = 'normal'
+            else:
+                self._mode = 'drive_winding'
+
+        if self._mode == 'normal':
+            if code == 'right_trigger':
+                self.set_speed(position)
+                self.drive()
+            if code == 'left_trigger':
+                self.set_speed(position)
+                self.reverse()
+            if code == 'dpad_left_right' or code == 'left_stick_x':
+                self.turn(code, position)
+
+        if self._mode == 'drives_in_circle':
```

Comment:
Can't we just do something like:

as long as we don't press another button, call this function. When we press another button, call the function that corresponds to that new button.